Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```diff
@@ -0,0 +1,273 @@
+
+        let osSuffixes = this.detectMachineOS();
+        let downloadPackageInfoObject: any = versionInfo.files.find((downloadPackageInfo: any) => {
+            if (downloadPackageInfo.rid.toLowerCase() == osSuffixes[0].toLowerCase()
+                && (osSuffixes[0].split("-")[0] == "win" && downloadPackageInfo.name.endsWith(".zip"))) {
+                return true;
+            }
+
+            return false;
+        });
+
+        if (!!downloadPackageInfoObject && downloadPackageInfoObject.url) {
+            return downloadPackageInfoObject.url;
+        }
+
+        return "";
+    }
+
+    private setReleasesIndex(): Promise<any> {
+        return this.httpCallbackClient.get(DotNetCoreReleasesIndexUrl)
+            .then((response: HttpClientResponse) => {
+                return response.readBody();
+            })
+            .then((body: string) => {
+                this.releasesIndex = JSON.parse(body);
+                return this.releasesIndex;
+            })
+            .catch((ex) => {
+                throw tl.loc("");
+            });
+    }
+
+    private getVersionChannel(version: string): any {
+        let versionParts: any = utils.getVersionParts(version);
+
+        let channelVersion = `${versionParts.majorVersion}.${versionParts.minorVersion}`;
+        if (versionParts.minorVersion == "x") {
+            var latestChannelVersion: string = `${versionParts.majorVersion}.0`;
+            this.releasesIndex["releases-index"].forEach(channel => {
+                if (utils.versionCompareFunction(channel["channel-version"], latestChannelVersion) > 0) {
+                    latestChannelVersion = channel["channel-version"];
+                }
+            });
+
+            channelVersion = latestChannelVersion;
+        }
+
+        return this.releasesIndex["releases-index"].find(element => {
+            if (element["channel-version"] == channelVersion) {
+                return true
+            }
+        });
+    }
+
+    private getVersionFromChannel(channelInformation: any, version: string, packageType: string, includePreviewVersions: boolean): Promise<VersionInfo> {
+        let versionParts: any = utils.getVersionParts(version);
+        var releasesJsonUrl: string = channelInformation["releases.json"];
+        var channelReleases: any = null;
+
+        if (releasesJsonUrl) {
+            return this.httpCallbackClient.get(releasesJsonUrl)
+                .then((response: HttpClientResponse) => {
+                    return response.readBody();
+                })
+                .then((body: string) => {
+                    channelReleases = JSON.parse(body).releases;
+                    if (versionParts.minorVersion == "x" || versionParts.patchVersion == "x") {
+                        let dotNetSdkVersionTelemetry = `{"version":"${version}"}`;
+                        console.log("##vso[telemetry.publish area=TaskDeploymentMethod;feature=DotNetCoreInstallerV1]" + dotNetSdkVersionTelemetry);
```

Comment:
why are we sending telemetry only in this case?

Reply:
I was thinking of only monitoring as to how many people are using the tasks with wildcards in version. Changed it now to always log user input version and resolved version